src/main/java/net/minecraftforge/common/capabilities/CapabilityProvider.java

```diff
@@ -48,17 +48,17 @@ protected final void gatherCapabilities(@Nullable ICapabilityProvider parent)
         this.capabilities = ForgeEventFactory.gatherCapabilities(baseClass, this, parent);
     }
 
-    protected final @Nullable CapabilityDispatcher getCapabilities()
+    protected @Nullable CapabilityDispatcher getCapabilities()
```

Comment:
Not sure if this should be changed or not but part of me wonders from an API standpoint and modders potentially doing weird things in their tiles among other things if these methods should stay final and then have helper package private methods be defined these proxy via and then those get overrode in `LazyCapabilityProvider`. That way modders can't start overriding them and potentially breaking them.

Reply:
Not sure tbh, seems quite involved, and modders can still mess it up in other ways. For example they can spam `gatherCapabilities`, or do weird stuff in the event.

Reply:
Changed the approach, now everything is final.